src/pyTypeFactory.cc

```diff
@@ -167,26 +139,15 @@ PyType *pyTypeFactory(JSContext *cx, JS::Rooted<JSObject *> *thisObj, JS::Rooted
         }
       }
     }
-    return new DictType(cx, *rval);
+    return new DictType(cx, rval);
   }
-  else if (rval->isMagic()) {
+  else if (rval.isMagic()) {
     printf("magic type is not handled by PythonMonkey yet\n");
   }
 
   std::string errorString("pythonmonkey cannot yet convert Javascript value of: ");
-  JS::RootedString str(cx, JS::ToString(cx, *rval));
+  JS::RootedString str(cx, JS::ToString(cx, rval));
   errorString += JS_EncodeStringToUTF8(cx, str).get();
   PyErr_SetString(PyExc_TypeError, errorString.c_str());
   return NULL;
-}
-
-PyType *pyTypeFactorySafe(JSContext *cx, JS::Rooted<JSObject *> *thisObj, JS::Rooted<JS::Value> *rval) {
-  PyType *v = pyTypeFactory(cx, thisObj, rval);
-  if (PyErr_Occurred()) {
-    // Clear Python error
-    PyErr_Clear();
-    // Return `pythonmonkey.null` on error
-    return new NullType();
-  }
-  return v;
```

Comment:
`pyTypeFactorySafe` exists because there's a `jsTypeFactorySafe`.

Reply:
Perhaps it was seen as good idea to have such symmetry in the past or in the early stages, however, unused methods are now gone and can be brought back eventually if needed